### ISSUE TITLE
[bug] [@eslint-react/naming-convention/use-state] does not enforce symmetric names

### ISSUE THREAD

**Issue by danielrentz** [Rel1cx/eslint-react#858]
### Describe the bug

[Documentation](https://eslint-react.xyz/docs/rules/naming-convention-use-state) states to check symmetric naming, but
```js
const [state, setValue] = useState(0);
```
passes.

### Expected behavior

Fails because of "state" vs. "setValue" naming mismatch, as `react/hook-use-state` does.

### Platform and versions

```Text
eslint: 9.15.0
@eslint-react/eslint-plugin: 1.16.1
```
